### PR TITLE
Update syn related dependencies to 1.0

### PR DESCRIPTION
**Pull request by Eijebong**

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -114,6 +122,14 @@ dependencies = [
  "proc-macro2 0.4.30 (registry+https://github.com/rust-lang/crates.io-index)",
 ]
 
+[[package]]
+name = "quote"
+version = "1.0.0"
```
Comment (emilio) [inline Cargo.lock]: Hmm, so we have two versions of quote and proc-macro2? That seems unfortunate.

Comment (Eijebong) [reply Cargo.lock]: https://github.com/serde-rs/serde/pull/1591 Unfortunately, serde isn't yet up to date :/

**Comment by emilio** [comment]
@dtolnay is https://github.com/serde-rs/serde/pull/1591 expected to land soon-ish? If so we may as well wait for that.

**Comment by dtolnay** [comment]
Done

File: Cargo.lock (hunk 1 above)
Comment (Eijebong) [reply Cargo.lock]: There, dupe removed :tada:

**Comment by emilio** [comment]
Thanks @dtolnay!

@Eijebong travis is complaining, mind taking a look?

**Comment by Eijebong** [comment]
Travis is always complaining :smile: 
Also looks like I failed at rebasing somehow. I'll fix it later today or tomorrow

**Comment by Eijebong** [comment]
There, should be fixed

File: src/bindgen/ir/cfg.rs [hunk 2]
```diff
@@ -136,48 +134,52 @@ impl Cfg {
...
     }
 
     fn load_single(item: &syn::NestedMeta) -> Option<Cfg> {
-        match *item {
-            syn::NestedMeta::Meta(syn::Meta::Word(ref ident)) => {
-                Some(Cfg::Boolean(format!("{}", ident)))
-            }
-            syn::NestedMeta::Meta(syn::Meta::NameValue(syn::MetaNameValue {
-                ref ident,
+        match item {
```
Comment (emilio) [inline src/bindgen/ir/cfg.rs]: Can you preserve the match style?

```rust
match *item {
    syn::NestedMeta::Meta(..) => { .. }
}
```

File: src/bindgen/ir/constant.rs [hunk 3]
```diff
@@ -160,29 +160,20 @@ impl Literal {
...
-                | syn::IntSuffix::I16
-                | syn::IntSuffix::I32
-                | syn::IntSuffix::I64
-                | syn::IntSuffix::I128 => unsafe {
+                "isize" | "i8" | "i16" | "i32" | "i64" | "i128" => unsafe {
                     Ok(Literal::Expr(format!(
                         "{}",
-                        mem::transmute::<u64, i64>(value.value())
+                        mem::transmute::<u64, i64>(
+                            value.base10_parse::<u64>().expect("Wrong number literal")
```
Comment (emilio) [inline src/bindgen/ir/constant.rs]: Can we use `base10_parse::<i64>` directly now?

Comment (emilio) [reply src/bindgen/ir/constant.rs]: Or just base10_digits everywhere? if not, why not?

File: src/bindgen/parser.rs [hunk 4]
```diff
@@ -318,33 +318,42 @@ impl<'a> Parser<'a> {
...
-                            })) = attr.interpret_meta()
-                            {
-                                match lit {
-                                    syn::Lit::Str(ref path) if ident == "path" => {
-                                        path_attr_found = true;
-                                        self.parse_mod(pkg, &mod_dir.join(path.value()))?;
-                                        break;
-                                    }
+                        let next_mod_path1 = mod_dir.join(next_mod_name.clone() + ".rs");
+                        let next_mod_path2 = mod_dir.join(next_mod_name.clone()).join("mod.rs");
```
Comment (emilio) [inline src/bindgen/parser.rs]: This looks like a rebase mistake? The logic is the same as a couple lines before.

File: src/bindgen/utilities.rs [hunk 5]
```diff
@@ -146,10 +146,9 @@ impl_syn_item_helper!(syn::ItemTrait);
 impl_syn_item_helper!(syn::ItemImpl);
 impl_syn_item_helper!(syn::ItemMacro);
 impl_syn_item_helper!(syn::ItemMacro2);
-impl_syn_item_helper!(syn::ItemExistential);
 impl_syn_item_helper!(syn::ItemTraitAlias);
 
-impl SynItemHelpers for syn::ItemVerbatim {
+/*impl SynItemHelpers for syn::ItemVerbatim {
```
Comment (emilio) [inline src/bindgen/utilities.rs]: If this no longer exists the code should be removed.

File: src/bindgen/utilities.rs [hunk 6]
```diff
@@ -83,9 +83,9 @@ macro_rules! syn_item_match_helper {
             syn::Item::Type(ref item) => (|$i: &syn::ItemType| $a)(item),
             syn::Item::Union(ref item) => (|$i: &syn::ItemUnion| $a)(item),
             syn::Item::Use(ref item) => (|$i: &syn::ItemUse| $a)(item),
-            syn::Item::Existential(ref item) => (|$i: &syn::ItemExistential| $a)(item),
             syn::Item::TraitAlias(ref item) => (|$i: &syn::ItemTraitAlias| $a)(item),
             syn::Item::Verbatim(_) => (|| $b)(),
+            _ => panic!("Unhandled syn::Item:  {:?}", $s),
```
Comment (emilio) [inline src/bindgen/utilities.rs]: I'd prefer to keep handling the others.

Comment (Eijebong) [reply src/bindgen/utilities.rs]: Unfortunately https://docs.rs/syn/1.0.3/src/syn/item.rs.html#79

File: src/bindgen/utilities.rs (hunk 5 above)
Comment (Eijebong) [reply src/bindgen/utilities.rs]: Woops... Forgot to remove that

File: src/bindgen/ir/constant.rs (hunk 3 above)
Comment (Eijebong) [reply src/bindgen/ir/constant.rs]: Yeah, we can use base10_digits, I just thought this was more useful than it really was :p

File: src/bindgen/parser.rs (hunk 4 above)
Comment (Eijebong) [reply src/bindgen/parser.rs]: :man_facepalming:

**Comment by emilio** [comment]
Oh, also, mind not doing the Cargo.toml version bump? I hope to merge #377 soonish, though I may as well do two releases so no big deal I guess.

**Comment by Eijebong** [comment]
Sure, I'll remove it

**Pull request merged**